### ISSUE TITLE
Issues with queries and comments

### ISSUE THREAD

**Issue by halmai** [HeidiSQL/HeidiSQL#162]
## Steps to reproduce this issue

1. Step 1; connect to a database
2. Step 2; create a query in the query text area plus add a commented line. Example:
    ```
    select now();
    -- dhsj
    ```` 
3. Step N; execute the query a couple of times by pressing F9 again and again
4. Then I get  an error message screen saying "An error occurred in the application"

## Current behavior

The erorr message appears.

## Expected behavior

Instead of the error message, the query should be executed.

## Environment

I think everything is there within the bugreport.txt file

**Comment by halmai** [comment]
[bugreport.txt](https://github.com/HeidiSQL/HeidiSQL/files/1758562/bugreport.txt)

**Comment by rentalhost** [comment]
Can't reproduce that on last nightly on MariaDB. Maybe it is a Postgre related?

**Comment by G3ph4z** [comment]
@rentalhost Yes, its using `libpq.dll`. 

@halmai Ha már magyarok vagyunk. :)
Milyen postgresql-el próbálod? 9.x vagy 10.x?
32 vagy 64 bites?

**Comment by halmai** [comment]
@G3ph4z Helló. :)

select version();
PostgreSQL 10.1, compiled by Visual C++ build 1800, 64-bit

ps: congratulation for your first contribution :)

**Comment by G3ph4z** [comment]
I do not remember seeing a similar mistake, but I will test again tonight. 

ps: Thank you! :)

**Comment by G3ph4z** [comment]
Okey I can confirm the issue, reproduced the issue. Its only appears with the new version (10.x).
But if you copy the `libpq.dll` from `C:\Program Files\PostgreSQL\10\lib` then it will solve your problem. 

@ansgarbecker Can you update the libpg.dll in Heidi's installer? 

(What we got with Heidi 9.5 and on the right is the newer)
http://prtsc.denes.cloud/vmware_2018-02-27_20-08-42.png

I tested to backwards, and everything is working fine with `libpg.dll` version 10 and postgresql 9.x

Magyarul is le kell, hogy irjam, mert rettenetes az angolom:
Szóval a 9.5-ös heidivel a 9.5.2es pgsql lib-et telepiti fel, de a 9.6 és a 10 között vannak lényeges eltérések, csak másold ki a libpg.dll-t a postgresql könyvtárából a heidi mappájába, és menni fog. :)

**Comment by halmai** [comment]
Thanks, it sounds reasonable, I will try it when I get home.

If this is the case, I mean, if HeidiSql uses his (hm.... if it is called Heidi then "her", not "his":)) own libpq.dll then it can cause version conflicts, I think. For example, after upgrading postgres, heidisql still wants to use the old one, or if heidisql is updated then it uses one that the underlying postgres won't support.

Moreover, as much as I understand, this links heidisql to one particular (or a limited number of) versions of postgres which makes it hard/impossible to use heidisql when you connect to different versions of postgresql running on different remote servers. 

Would it be possible to do the following things:
- the installer should contain several versions of libpg.dll in order to pick the one that the current server needs
- the running code could detect that the given version of the server is not completely compatible with the current version of the libpq.dll so can tell the user that they should download the appropriate version of libpg.dll
- if the server host is "localhost" then heidisql should be able to find the appropriate libpq.dll in the local file system.

I admit, the messages can be annoying for the user but it still would be better than not working at all. 

The best would be if newer libpq.dll would totally support older versions of postgres as well. Then only the newest dll should be bundled into heidisql installer as soon as a new version comes out. This would solve the case when the user wants to connect to different version of postgres servers _at the same time_. Maybe this already happens. I should try it. :)

ps: megértettem az angolból is, nem olyan rettenetes az. :)

**Comment by G3ph4z** [comment]
its hard to follow the version changes. But your second idea is feasible, most of other commercial tools have similar function, like EMS, DeVart, etc....

I will check it tomorrow whether I can do it or not.

**Comment by mglatz** [comment]
...I see this on MariaDb using HeidySQL Portable 9.5.0.5196

does work but throws an error
```
select * from t_person;
--where ID = 10

```
does not work
```
select * from t_person
--where ID = 10

```

does not work
```
select * from t_person
--where ID = 10
;
```

**Comment by huyphams** [comment]
@mglatz  In MySQL (MariaDB), you must include at least one space after --

```
From a --  sequence to the end of the line. In MySQL, the --  (double-dash) comment style requires the second dash to be followed by at least one whitespace or control character (such as a space, tab, newline, and so on). This syntax differs slightly from standard SQL comment syntax, as discussed in Section 1.8.2.4, “'--' as the Start of a Comment”.
```

Link: https://dev.mysql.com/doc/refman/8.0/en/comments.html

**Comment by mglatz** [comment]
Ooooh well thanks :)

**Comment by ansgarbecker** [comment]
I am confused, due to too much time elapsed, and too many comments here, probably also unrelated ones.

Is there still an issue in HeidiSQL?

**Comment by ansgarbecker** [comment]
Closing with no feedback within months.
